### ISSUE TITLE
[RFC] How to call `session.query()` for RDB storages

### ISSUE THREAD

**Issue by HideakiImamura** [optuna/optuna#2092]
`storages.RDBStorage` and its variants access the DB by calling `session.query()`. Currently, access to the DB is wrapped by `storages/_rdb/models.py`, but some `session.query()` calls are written directly to `storages/_rdb/storage.py` to speed up DB access. Shouldn't the `session.query()` calls be in one file for consistency?

## Candidates
1.  Define methods of all `session.query()` calls in `storages/_rdb/models.py`.
[pros] Direct access to the DB is hidden from `storages/_rdb/storage.py` and the logic is common.
[cons] Most of the same forms of `session.query()` are called only once, so the amount of logic to be commonized is small. The structure of `storages/_rdb` becomes more complex.
2. Use all `session.query()` directly in `storages/_rdb/models.py`, and remove `storages/_rdb/models.py`
[pros] The structure of `storages/_rdb` will be simplified.
[cons] `storages/_rdb/storage.py` becomes long.
3. (Variant of 1) Define methods of all `session.query()` calls in one file as custom query classes.
[pros] In addition to the advantages of 1, the implementation of `storages/_rdb/models.py` itself becomes simpler.
[cons] Same as that of 1.
4. Prepare an API that outputs `FrozenTrial` with `trial_id` as input, and remove APIs not needed for this from storage. As a result, delete all unneeded methods from `models.py`. (This candidate can be combined with other above candidates.)
